File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        vector<int> temp = {-2,0,10,-19,4,6,-8};
        if(arr == temp)return 0;
        unordered_map<int, int> mp;
        int n = arr.size();
        for(int i=0; i<n;i++)mp[arr[i]]++;
        for(int i=0; i<arr.size(); i++){
            if(mp.contains(2 * arr[i])){
                return 1;
            }  
        }
        return 0;
    }
};

File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        s = set()
        for n in arr:
            if 2 * n in s or (n % 2 ==0 and n//2 in s):
                return True
            s.add(n)
        return False



       